boot/boot.go

```diff
@@ -455,31 +454,43 @@ func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *Boota
 		return fmt.Errorf("cannot write modeenv: %v", err)
 	}
 
-	// get the ubuntu-boot bootloader
+	ubuntuBootPartition := filepath.Join(runMnt, "ubuntu-boot")
+
+	// get the ubuntu-boot grub and extract the kernel there
 	opts := &bootloader.Options{
 		// TODO:UC20: we use "recovery: true" here because on
 		// the partition the file layout of ubuntu-boot looks
 		// the same as ubuntu-seed.
 		// TODO:UC20: need a better name than recovery
 		Recovery: true,
+		// extract kernel assets to the ubuntu-boot partition
+		KernelExtractionDir: ubuntuBootPartition,
```

Comment:
it's unclear to me why we need this param, is the same dir we find the bootloader in? what am I missing?

Reply:
for uc20, we have to extract the kernel assets to "/run/mnt/ubuntu-boot/kernel.efi" whereas with uc18 we were extracting the kernel assets to `/boot/grub/ubuntu-kernel_42.snap/kernel.img` for images that needed full disk encryption (as indicated by the presence of a file in the kernel snap, see 

https://github.com/snapcore/snapd/blob/1d90e43ebf99f32b83f5fe1abb593162b8fa1b0a/bootloader/grub.go#L135-L141

does that make sense?

Reply:
for uc20, we have to extract the kernel assets to "/run/mnt/ubuntu-boot/kernel.efi" whereas with uc18 we were extracting the kernel assets to `/boot/grub/ubuntu-kernel_42.snap/kernel.img` for images that needed full disk encryption (as indicated by the presence of a file in the kernel snap, see 

https://github.com/snapcore/snapd/blob/1d90e43ebf99f32b83f5fe1abb593162b8fa1b0a/bootloader/grub.go#L135-L141

does that make sense?

Reply:
I forgot that we put them in /grub for uc18/16. Either way we do still want the "kernel_42.snap" bit in the extraction paths, whatever the core system, and then we want a symlink  from [grub/]kernel.efi to the <kernel>_<REVNO>.snap/kernel.efi in play. The closest extraction works between uc18/16 vs uc20 the better.  Any difference should be taken care with symlinks.

Reply:
to be clear the symlinks should be a problem for MakeBootable, SetNextBoot, not the extraction code

Reply:
Okay, I added code to `makeBootable20RunMode` to create a symlink and the kernel.efi is now actually extracted to the same uc18/uc16 location, but note that it is extracted to ubuntu-boot partition, which isn't exactly the same as the uc16/uc18 layout, as uc20 has `/EFI/ubuntu` while uc16/18 has `/boot/grub`.